### PR TITLE
getIndex filter

### PR DESCRIPTION
**Pull request by tobibeer**
As discussed at #1375, respectively https://github.com/Jermolene/TiddlyWiki5/pull/1375#issuecomment-71909477, for a data tiddler, returns the value at the index specified in the operand.

`<$list filter="[[SomeData]getIndex[someIndex]]"/>`

**demo:** http://tbdemo.tiddlyspot.com/#Index%20Filter

### CODE REVIEW

File: core/modules/filters/getIndex.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+/*\
+title: $:/core/modules/filters/getIndex.js
```
Comment (Jermolene) [inline core/modules/filters/getIndex.js]: It's inconsistent to use "getIndex"; all the other two word filters just concatenate lower case words (eg, "removeprefix", "moduletypes"). The filename needs changing too.

File: core/modules/filters/getIndex.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+/*
+Export our filter function
+*/
+exports.getIndex = function(source,operator,options) {
+	var data,title,results = [];
+	if(operator.operand){
+		source(function(tiddler,title) {
+			title = tiddler ? tiddler.fields.title : title;
+			data = options.wiki.parseTextReference(title,"",operator.operand,{});
```
Comment (Jermolene) [inline core/modules/filters/getIndex.js]: this doesn't look quite right. `parseTextReference` parses the content of a field or index of a tiddler and returns a parse tree. It does return the source text in the `source` property, but that's really a side effect of the parsing process. It's not an efficient way to access a property index. Better to use `wiki.extractTiddlerDataItem()`.

**Comment by Jermolene** [comment]
Looks good - @aelocson are the docs changes OK?

File: core/modules/filters/getIndex.js (hunk 2 above)
Comment (tobibeer) [inline core/modules/filters/getIndex.js]: thanks, I was actually wondering about that and meaning to ask ...pushing an update

File: core/modules/filters/getIndex.js (hunk 1 above)
Comment (tobibeer) [inline core/modules/filters/getIndex.js]: thanks, changed

**Comment by tobibeer** [comment]
code, docs and demo updated as suggested

**Comment by Jermolene** [comment]
Thanks @tobibeer

**Comment by BramChen** [comment]
```javascript
exports.getIndex = function(source,operator,options) {
...
};
```

It should be **getindex**.

**Comment by tobibeer** [comment]
@BramChen, thanks for being watchful!

**Pull request merged**